.github/workflows/main.yml

```
@@ -1,21 +1,86 @@
-on: [push]
+name: Tests
+on:
+  pull_request:
+  push:
+    branches:
+      - main
 
 jobs:
-  new_push_job:
+  unit_tests:
     runs-on: ubuntu-latest
-    name: New push to repo
     steps:
-    - name: Publish to slack
+    - uses: actions/checkout@v2
+    - run: npm ci && npm run build
+    - run: npm test
+
+  integration_test_botToken:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - run: npm ci && npm run build
+    - name: Post message to Slack via botToken
       id: slack
-      uses: slackapi/slack-github-action@v1.15.0
+      uses: ./
       with:
-        channel-id: 'CTAAHRA79'
-        slack-message: 'posting from a github action!'
-        payload: "{\"key\":\"value\",\"foo\":\"bar\"}"
+        channel-id: ${{ secrets.SLACK_CHANNEL_ID }}
+        slack-message: 'CI Post from slack-send GitHub Action! Succeeded!!'
+        # payload: "{\"key\":\"value\",\"foo\":\"bar\"}"
       env:
         SLACK_BOT_TOKEN: ${{ secrets.SLACK_BOT_TOKEN }}
+
+    # Use the output from the `hello` step
+    - name: Get the output time
+      run: echo "The time was ${{ steps.slack.outputs.time }}"
```

Instead of just echoing the time, can we check that it's a non-zero length string? This way if the output is broken somehow, we'd be notified in CI. Something like:

```suggestion
      run: test -n "${{ steps.slack.outputs.time }}"
```